packages/consensus/src/types/utxo_set.cairo

```diff
@@ -31,22 +35,29 @@ pub impl UtxoSetImpl of UtxoSetTrait {
         UtxoSet { utreexo_state, ..Default::default() }
     }
 
-    fn add(ref self: UtxoSet, output: OutPoint) {
+    fn add(ref self: UtxoSet, output: OutPoint) -> Result<(), ByteArray> {
         if output.data.cached {
-            let outpoint_hash = PoseidonTrait::new().update_with(output).finalize();
-            self.cache.insert(outpoint_hash, true);
+            let hash = PoseidonTrait::new().update_with(output).finalize();
```

Comment:
Don't we want Outpoint to have a `hash` method? `PoseidonTrait::new().update_with` is being repeated all over the place?

Reply:
Yeah, we will need to refactor it, after this PR and Utreexo add/remove are merged would be better I think.
Note that `Outpoint::hash` will also require additional `block_hash` argument, not sure if it should be a struct method rather than a free function.

Reply:
Follow up https://github.com/keep-starknet-strange/raito/issues/205